docs/Conditions.md

```diff
@@ -191,4 +211,50 @@ Usage can then look as following:
 #else
 // This renders for desktop platforms
 #endif
-```
\ No newline at end of file
+```
+
+## Conditional Parameters
+
+[Parameter symbols in template](Reference-for-template.json.md#parameter-symbol) can be specified together with optional conditions:
+* [`IsEnabled Condition`](Reference-for-template.json.md#isEnabled) - overwritting presence of input parameter. If condition is specified and evaluates to false (or a `false` constant is passed), passed parameter value (if any) is ignored and processing works as if the parameter value was not passed. This includes application of [default values](Reference-for-template.json.md#default), [verification of mandatory parameters](Reference-for-template.json.md#isRequired), [conditional processing of sources](Conditional-processing-and-comment-syntax.md) and [replacements](Reference-for-template.json.md#replaces).
+* [`IsRequired Condition`](Reference-for-template.json.md#isRequired) - dictates if parameter is mandatory or optional.
+
+### Evaluation
+
+**Input** - currently only other parameter symbols from the template configuration are supported within the parameter conditions. Any other variables are not bound and replaced (they are considered part of literal string).
+
+**Evaluation order** - Dependencies between parameters are detected and evaluation is peformed in order that guarantees that all dependencies are evaluated prior their dependant (see [Topological Sorting](https://en.wikipedia.org/wiki/Topological_sorting) for details).
+
+ In case of cyclic dependency the evaluation proceeds only if current input values of parameters do not lead to indeterministic result (and the cycle is indicated in warning log message). Otherwise an error is reported, indicating the cycle.
+
+ Example `template.json` with cyclic dependency:
+```json
+  "symbols": {
+    "A": {
+      "type": "parameter",
+      "datatype": "bool",
+      "isEnabled": "B != false",
+    },
+    "B": {
+      "type": "parameter",
+      "datatype": "bool",
+      "isEnabled": "A != true",
+    }
+}
+```
+
+Following input parameter values can (and will) be evaluated deterministically: `--A false --B true`
+
+Following input parameter values cannot be evaluated deterministically (and will lead to error): `--A true --B false`
+
+**Applying user, host and default values** - All user passed, host and default values are applied before the conditions evaluation. After the evaluation defaults are reapplied to parameters that were evaluated as optional and that do not have user/host supplied values. After this an evaluation of presence of mandatory values takes place.
+
+### Performing evaluation externally
```

Comment:
Just wondering if this paragraph should be added to api folder instead.

Reply:
Good idea - moving there